parquet-variant-compute/src/shred_variant.rs

```diff
@@ -0,0 +1,717 @@
+        let temp_array = create_test_variant_array(vec![Some(Variant::from("test"))]);
+        let metadata = temp_array.metadata_field().clone();
+        let value = temp_array.value_field().unwrap().clone();
+        let typed_value = Arc::new(Int64Array::from(vec![42])) as ArrayRef;
+
+        let shredded_array =
+            VariantArray::from_parts(metadata, Some(value), Some(typed_value), None);
+
+        let result = shred_variant(&shredded_array, &DataType::Int64);
+        assert!(matches!(
+            result.unwrap_err(),
+            ArrowError::InvalidArgumentError(_)
+        ));
+    }
+
+    #[test]
+    fn test_all_null_input() {
+        // Create VariantArray with no value field (all null case)
+        let metadata = BinaryViewArray::from_iter_values([&[1u8, 0u8]]); // minimal valid metadata
+        let all_null_array = VariantArray::from_parts(metadata, None, None, None);
+        let result = shred_variant(&all_null_array, &DataType::Int64).unwrap();
+
+        // Should return array with no value/typed_value fields
+        assert!(result.value_field().is_none());
+        assert!(result.typed_value_field().is_none());
+    }
+
+    #[test]
+    fn test_unsupported_list_schema() {
+        let input = create_test_variant_array(vec![Some(Variant::from(42))]);
+        let list_schema = DataType::List(Arc::new(Field::new("item", DataType::Int64, true)));
+        shred_variant(&input, &list_schema).expect_err("unsupported");
+    }
+
+    #[test]
+    fn test_primitive_shredding_comprehensive() {
+        // Test mixed scenarios in a single array
+        let input = create_test_variant_array(vec![
+            Some(Variant::from(42i64)),   // successful shred
+            Some(Variant::from("hello")), // failed shred (string)
+            Some(Variant::from(100i64)),  // successful shred
+            None,                         // array-level null
+            Some(Variant::Null),          // variant null
+            Some(Variant::from(3i8)),     // successful shred (int8->int64 conversion)
+        ]);
+
+        let result = shred_variant(&input, &DataType::Int64).unwrap();
+        println!("result: {:?}", result);
+
+        // Verify structure
+        let value_field = result.value_field().unwrap();
+        let typed_value_field = result
+            .typed_value_field()
+            .unwrap()
+            .as_any()
+            .downcast_ref::<Int64Array>()
+            .unwrap();
+
+        // Check specific outcomes for each row
+        assert_eq!(result.len(), 6);
+
+        // Row 0: 42 -> should shred successfully
+        assert!(!result.is_null(0));
+        assert!(value_field.is_null(0)); // value should be null when shredded
+        assert!(!typed_value_field.is_null(0));
+        assert_eq!(typed_value_field.value(0), 42);
+
+        // Row 1: "hello" -> should fail to shred
+        assert!(!result.is_null(1));
+        assert!(!value_field.is_null(1)); // value should contain original
```

Comment:
I recommend also  verify the non shredded value, something like

```suggestion
        assert!(!value_field.is_null(1)); // value should contain original
        assert_eq!(result.value(1), Variant::from("hello"));
```